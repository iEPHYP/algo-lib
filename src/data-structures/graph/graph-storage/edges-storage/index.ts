export * from './edges-storage.types';
export * from './edges-storage.utils';
