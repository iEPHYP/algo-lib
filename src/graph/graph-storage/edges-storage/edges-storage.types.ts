export interface Edge<TVertex = number> {
  vertexOutIndex: TVertex;
  vertexInIndex: TVertex;
}

export type EdgesStoredGraph = Edge[];

export type RawEdges = [number, number][];
