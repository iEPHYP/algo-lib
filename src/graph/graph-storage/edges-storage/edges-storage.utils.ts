import { RawEdges } from './edges-storage.types';

export const getNormalizedEdges = (rawEdges: RawEdges) =>
  rawEdges.map(([vertexOutIndex, vertexInIndex]) => ({
    vertexOutIndex,
    vertexInIndex,
  }));
