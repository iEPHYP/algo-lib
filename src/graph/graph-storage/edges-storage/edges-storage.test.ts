import { rawEdges } from '../../rawEdges.mock';
import { getNormalizedEdges } from './edges-storage.utils';

test('getNormalizedEdges properly converts Raw Edges into Normalized Edges', () => {
  const normalizedEdges = getNormalizedEdges(rawEdges);

  expect(normalizedEdges).toMatchSnapshot();
});
