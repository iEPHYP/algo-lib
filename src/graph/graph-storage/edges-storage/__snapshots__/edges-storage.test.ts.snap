// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNormalizedEdges properly converts Raw Edges into Normalized Edges 1`] = `
[
  {
    "vertexInIndex": 1,
    "vertexOutIndex": 0,
  },
  {
    "vertexInIndex": 3,
    "vertexOutIndex": 0,
  },
  {
    "vertexInIndex": 4,
    "vertexOutIndex": 3,
  },
  {
    "vertexInIndex": 7,
    "vertexOutIndex": 4,
  },
  {
    "vertexInIndex": 5,
    "vertexOutIndex": 1,
  },
  {
    "vertexInIndex": 6,
    "vertexOutIndex": 1,
  },
  {
    "vertexInIndex": 6,
    "vertexOutIndex": 5,
  },
  {
    "vertexInIndex": 2,
    "vertexOutIndex": 6,
  },
]
`;
